src/Authentication/Authentication.Test/Helpers/HttpHelpersTests.cs

```diff
@@ -71,5 +72,112 @@ public void GetGraphHttpClientShouldReturnHttpClientWithCustomerProvidedTimeout(
             // reset static instance.
             GraphSession.Reset();
         }
+
+        [Fact]
+        public void GetGraphHttpClientShouldReturnInSessionHttpClientWhenSessionHasAClient()
+        {
+            GraphSession.Initialize(() => new GraphSession());
+            GraphSession.Instance.AuthContext = new AuthContext
+            {
+                AuthType = AuthenticationType.UserProvidedAccessToken,
+                ContextScope = ContextScope.Process,
+            };
+            GraphSession.Instance.GraphHttpClient = new HttpClient
+            {
+                BaseAddress = new Uri("https://test.contoso.com/v1.0/")
+            };
+
+            HttpClient httpClient = HttpHelpers.GetGraphHttpClient();
+
+            Assert.NotNull(httpClient);
+            Assert.NotNull(GraphSession.Instance.GraphHttpClient);
+            Assert.Equal(httpClient.BaseAddress, GraphSession.Instance.GraphHttpClient.BaseAddress);
+
+            // reset static instance.
+            GraphSession.Reset();
+        }
+
+        [Fact]
+        public void GetGraphHttpClientShouldReturnNewHttpClientWhenSessionHasNoClient()
+        {
+            GraphSession.Initialize(() => new GraphSession());
+            GraphSession.Instance.AuthContext = new AuthContext
+            {
+                AuthType = AuthenticationType.UserProvidedAccessToken,
+                ContextScope = ContextScope.Process,
+            };
+            GraphSession.Instance.GraphHttpClient = null;
+
+            HttpClient httpClient = HttpHelpers.GetGraphHttpClient();
+
+            Assert.NotNull(httpClient);
+            Assert.NotNull(GraphSession.Instance.GraphHttpClient);
+            Assert.Equal(httpClient.BaseAddress, GraphSession.Instance.GraphHttpClient.BaseAddress);
+
+            // reset static instance.
+            GraphSession.Reset();
+        }
+
+        [Fact]
+        public void GetGraphHttpClientShouldReturnNewHttpClientWhenSessionIsNew()
+        {
+            GraphSession.Initialize(() => new GraphSession());
+            GraphSession.Instance.AuthContext = new AuthContext
+            {
+                AuthType = AuthenticationType.UserProvidedAccessToken,
+                ContextScope = ContextScope.Process,
+            };
+
+            HttpClient httpClient = HttpHelpers.GetGraphHttpClient();
+
+            Assert.NotNull(httpClient);
+            Assert.NotNull(GraphSession.Instance.GraphHttpClient);
+            Assert.Equal(httpClient.BaseAddress, GraphSession.Instance.GraphHttpClient.BaseAddress);
+
+            // reset static instance.
+            GraphSession.Reset();
+        }
+
+        [Fact]
+        public void GetGraphHttpClientShouldReturnNewHttpClientSignOutThenSignIn()
```

Comment:
shouldn't there be a test that win disconnect-mggraph is done the GraphHttpClient instance is null.

Reply:
I'm already mocking disconnect-mggraph in https://github.com/microsoftgraph/msgraph-sdk-powershell/pull/1184/files/adb98883d4da7d3925cad95307e0e36f4ff9dbac#diff-33c29c1e330c4609e494c23e4a071b20d7e67bedd59120cd21d2511b69fe08c3R159-R161, by explicitly setting AuthContext and GraphHTTPClient to null. We don't have a way of invoking disconnect-mggraph directly in an Xunit test.

Reply:
I'll introduce pester tests in https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/1144 to cover invoking of commands in a powershell process.